File: go/internal/xiansword/xiansword.go
package xiansword

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

type SwordEnum int

const (
	Gold SwordEnum = iota
	Wood
	Water
	Fire
	Soil
)

func (s SwordEnum) String() string {
	switch s {
	case Gold:
		return "金剑"
	case Wood:
		return "木剑"
	case Water:
		return "水剑"
	case Fire:
		return "火剑"
	case Soil:
		return "土剑"
	default:
		return "未知剑"
	}
}

func SwordToString(enumA SwordEnum) string {
	return enumA.String()
}

type SwordException struct {
	msg string
}

func (s SwordException) Error() string {
	return s.msg
}

func NewSwordException(err string) SwordException {
	return SwordException{msg: err}
}

type SwordInFile struct {
	file string
}

func NewSwordInFile(file1 string) *SwordInFile {
	file := &SwordInFile{file: file1}
	file.ensureFileExists()
	return file
}

func (s *SwordInFile) ensureFileExists() {
	if _, err := os.Stat(s.file); os.IsNotExist(err) {
		file, err := os.Create(s.file)
		if err == nil {
			file.Close()
		}
	}
}

func (s *SwordInFile) Get() SwordEnum {
	data, err := os.ReadFile(s.file)
	if err != nil {
		return Gold
	}
	
	content := strings.TrimSpace(string(data))
	if content == "" {
		return Gold
	}
	
	tmp2, err := strconv.Atoi(content)
	if err != nil {
		return Gold
	}
	
	if tmp2 < 0 || tmp2 > 4 {
		return Gold
	}
	
	return SwordEnum(tmp2)
}

func (s *SwordInFile) Set(thing SwordEnum) error {
	tmp := int(thing)
	
	file, err := os.Create(s.file)
	if err != nil {
		return NewSwordException(fmt.Sprintf("无法创建文件: %v", err))
	}
	defer file.Close()
	
	_, err = file.WriteString(strconv.Itoa(tmp))
	if err != nil {
		return NewSwordException(fmt.Sprintf("写入文件失败: %v", err))
	}
	
	return nil
}

File: lib/logic/xiansword.dart
import 'dart:io';

enum SwordEnum {
  gold,
  wood,
  water,
  fire,
  soil;

  @override
  String toString() {
    switch (this) {
      case SwordEnum.gold:
        return "金剑";
      case SwordEnum.wood:
        return "木剑";
      case SwordEnum.water:
        return "水剑";
      case SwordEnum.fire:
        return "火剑";
      case SwordEnum.soil:
        return "土剑";
    }
  }

  static SwordEnum fromInt(int value) {
    if (value < 0 || value > 4) {
      return SwordEnum.gold;
    }
    return SwordEnum.values[value];
  }
}

String swordToString(SwordEnum enumA) {
  return enumA.toString();
}

class SwordException implements Exception {
  final String msg;

  SwordException(this.msg);

  @override
  String toString() => msg;
}

SwordException newSwordException(String err) {
  return SwordException(err);
}

class SwordInFile {
  final String file;

  SwordInFile(this.file) {
    _ensureFileExists();
  }

  factory SwordInFile.create(String file1) {
    return SwordInFile(file1);
  }

  void _ensureFileExists() {
    File f = File(file);
    if (!f.existsSync()) {
      try {
        f.createSync();
      } catch (e) {
        // 忽略创建文件时的错误
      }
    }
  }

  SwordEnum get() {
    try {
      File f = File(file);
      if (!f.existsSync()) {
        return SwordEnum.gold;
      }

      String data = f.readAsStringSync();
      String content = data.trim();
      if (content.isEmpty) {
        return SwordEnum.gold;
      }

      int value = int.tryParse(content) ?? -1;
      if (value < 0 || value > 4) {
        return SwordEnum.gold;
      }

      return SwordEnum.fromInt(value);
    } catch (e) {
      return SwordEnum.gold;
    }
  }

  Future<void> set(SwordEnum thing) async {
    try {
      File f = File(file);
      await f.writeAsString(thing.index.toString());
    } catch (e) {
      throw newSwordException("写入文件失败: $e");
    }
  }
}